### PR TITLE
Fix for greedy autostartland

### PR DESCRIPTION
**Pull request by plaursen**
Improved filtering for selecting candidate flights for autostart/autoland

### CODE REVIEW

File: FlightJournal.Web/Aprs/AircraftEventHandler.cs [hunk 1]
```diff
@@ -24,14 +24,15 @@ public AircraftEventHandler(IAprsListener aprsListener, FlightContext db)
...
+            Plane p = _db.Planes.FirstOrDefault(x => x.Registration.ToLower() == e.Aircraft.Registration.ToLower() 
+                                                     || x.CompetitionId.ToLower() == e.Aircraft.CompetitionId.ToLower()); // need comp. id due to non-stringent registrations...
             if (p == null)
             {
                 Log.Debug($"{nameof(AircraftEventHandler)}: starting {e.Aircraft} not in DB - ignored");
                 return;
             }
 
-            var flights = _db.Flights.Where(f => f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
+            var flights = _db.Flights.Where(f =>  !f.Deleted.HasValue && f.LastUpdated.Date == DateTime.Today && f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
```
Comment (janhebnes) [inline FlightJournal.Web/Aprs/AircraftEventHandler.cs:35]: We need to double check that the f.Deleted.HasValue element is rendered directly to SQL and does not result in full result set back (should be logged in trace view)

Comment (plaursen) [reply FlightJournal.Web/Aprs/AircraftEventHandler.cs:35]: or simply check on null instead, I guess

File: FlightJournal.Web/Aprs/AircraftEventHandler.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ private void OnAircraftTakeoff(object sender, AircraftEvent e)
                 return;
             }
 
-            var flights = _db.Flights.Where(f =>  !f.Deleted.HasValue && f.LastUpdated.Date == DateTime.Today && f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
+            var flights = _db.Flights.Where(f => f.Deleted != null && f.LastUpdated.Date == DateTime.Today && f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
```
Comment (janhebnes) [inline FlightJournal.Web/Aprs/AircraftEventHandler.cs:35]:
```suggestion
            var flights = _db.Flights.Where(f => f.Deleted == null && f.LastUpdated.Date == DateTime.Today && f.Plane.PlaneId == p.PlaneId && f.Departure == null && f.Landing == null);
```

File: FlightJournal.Web/Aprs/AircraftEventHandler.cs [hunk 3]
```diff
@@ -69,7 +69,7 @@ private void OnAircraftLanding(object sender, AircraftEvent e)
                 return;
             }
 
-            var flights = _db.Flights.Where(f => !f.Deleted.HasValue && f.Plane.PlaneId == p.PlaneId && f.Departure.HasValue && f.Departure.Value.Date == DateTime.Today && f.Landing == null);
+            var flights = _db.Flights.Where(f => f.Deleted != null && f.Plane.PlaneId == p.PlaneId && f.Departure!=null && f.Departure.Value.Date == DateTime.Today && f.Landing == null);
```
Comment (janhebnes) [inline FlightJournal.Web/Aprs/AircraftEventHandler.cs:72]:
```suggestion
            var flights = _db.Flights.Where(f => f.Deleted == null && f.Plane.PlaneId == p.PlaneId && f.Departure!=null && f.Departure.Value.Date == DateTime.Today && f.Landing == null);
```

**Comment by plaursen** [comment]
Ups.

**Review by janhebnes — APPROVED** [review]

**Pull request merged**